docs/source/en/migration.md

```diff
@@ -1 +1,300 @@
+```js
+const [ news, user ] = yield [
+  ctx.service.news.list(topic),
+  ctx.service.user.get(uid),
+];
+```
+
+In this case, use `Promise.all()` to wrap it:
+
+```js
+const [ news, user ] = await Promise.all([
+  ctx.service.news.list(topic),
+  ctx.service.user.get(uid),
+]);
+```
+
+#### object - yield {}
+
+Sometimes `yield {}` and `yield map` can also be used to send concurrent requests, but it may be a bit complex in this place because `Promise.all` doesn't support Object argument.
+
+```js
+// app/service/biz.js
+class BizService extends Service {
+  * list(topic, uid) {
+    return {
+      news: ctx.service.news.list(topic),
+      user: ctx.service.user.get(uid),
+    };
+  }
+}
+
+// app/controller/home.js
+const { news, user } = yield ctx.service.biz.list(topic, uid);
+```
+
+It's recommended to use `await Promise.all([])`:
+
+```js
+// app/service/biz.js
+class BizService extends Service {
+  list(topic, uid) {
+    return Promise.all([
+      ctx.service.news.list(topic),
+      ctx.service.user.get(uid),
+    ]);
+  }
+}
+
+// app/controller/home.js
+const [ news, user ] = await ctx.service.biz.list(topic, uid);
+```
+
+If the interfaces are unchangeable, we can temporarily do:
+
+- Use [app.toPromise] method provided by our Utils.
+- **This is built on top of the [co], so it may still cause performance issue and returning inaccurate error stacks, so this is not recommended.**
+
+```js
+const { news, user } = await app.toPromise(ctx.service.biz.list(topic, uid));
+```
+
+#### Others
+
+- thunks (functions)
+- generators (delegation)
+- generator functions (delegation)
+
+Use `async function` to replace the above functions, or use [app.toAsyncFunction] to wrap them if async function is not the case.
+
+**Note**
+- [toAsyncFunction][app.toAsyncFunction] and [toPromise][app.toPromise] are wrappers of [co], thus it may cause performance lost and error stack problems. So we're recommending developers to use all-chain upgrade.
+- [toAsyncFunction][app.toAsyncFunction] doesn't have lost when invokes async function.
+
+@sindresorhus has wrote a lot [promise-based helpers](https://github.com/sindresorhus/promise-fun), use them together with async function could make source code more readable.
+
+## Plugin update
+`App developers` just need to update the upgraded plugins by `plugin developers`, or use `egg-bin autod` command we've prepared to quickly update.
+
+The following content is for `plugin developers`, it shows how to update the plugins:
+
+### Update precautions
+
+- Finish the upgrade items above.
+  - Replace all `generator function` to `async function`.
+  - Upgrade middlewares.
+- Interfaces compabilities (optional), see following.
+- Release a major version.
+
+### Interfaces compabilities
+In some cases, the interface provided by `Plugin developers` supports both generator and async, normally it's wrapped by co.
+
+- In 2.x, we suggest to change to `async-first` in order to get a better performance and clearer error stacks.
```

Comment:
remove `to change to`

Reply:
Updated. 
I felt 
> we suggest to use `async-first` for a better performance and clearer error stacks.
could be more natural. But I'm sure either way people'll get the point.